Dalamud.DiscordBridge/DiscordHandler.cs

```diff
@@ -418,6 +482,49 @@ private async Task<bool> EnsureOwner(IUser user, ISocketMessageChannel errorMess
             return false;
         }
 
+        public async Task SendItemSaleEvent(uint itemId, int amount, bool isHq, string message, XivChatType chatType)
+        {
+            var applicableChannels =
+                this.plugin.Config.ChannelConfigs.Where(x => x.Value.ChatTypes.Contains(chatType));
+
+            if (!applicableChannels.Any())
+                return;
+
+            message = this.specialChars.TransformToUnicode(message);
+
+            var avatarUrl = Constant.LogoLink;
+            var itemName = String.Empty;
+            PluginLog.Information($"Retainer sold item: {itemId}");
+            try
+            {
+                ItemResult res = XivApiClient.GetItem(itemId).GetAwaiter().GetResult();
```

Comment:
Don't use XIVAPI for this, especially not in DiscordHandlers - I'd grab it from Lumina in DiscordMessageQueue

Reply:
Is there a better way to grab the icon? That's all I really needed xivapi for. This was largely copied from the old bot and I could adjust the function to have the MessageQueue send over the item name, cutting things down to really just a need for an icon. 

Based on some quick testing, the icon ushort that Lumina provides from an item doesn't match xivapi URLs, so I'm not sure if we can just hardcode one programmatically or if I'm just looking at the wrong thing.